pkg/mqtrigger/scalermanager.go

```diff
@@ -0,0 +1,265 @@
+package mqtrigger
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	fv1 "github.com/fission/fission/pkg/apis/core/v1"
+	"github.com/fission/fission/pkg/crd"
+	"github.com/fission/fission/pkg/utils"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	k8sCache "k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	kedaGVR = schema.GroupVersionResource{
+		Group:    "keda.k8s.io",
+		Version:  "v1alpha1",
+		Resource: "scaledobjects",
+	}
+)
+
+func getKedaClient(namespace string) (dynamic.ResourceInterface, error) {
+	var config *rest.Config
+	var err error
+
+	// get the config, either from kubeconfig or using our
+	// in-cluster service account
+	kubeConfig := os.Getenv("KUBECONFIG")
+	if len(kubeConfig) != 0 {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeConfig)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, err
+		}
+	}
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return dynamicClient.Resource(kedaGVR).Namespace(namespace), nil
+}
+
+func StartScalerManager(logger *zap.Logger, routerUrl string) error {
+	fissionClient, kubeClient, _, err := crd.MakeFissionClient()
+	if err != nil {
+		return err
+	}
+	err = fissionClient.WaitForCRDs()
+	if err != nil {
+		return errors.Wrap(err, "error waiting for CRDs")
+	}
+	crdClient := fissionClient.CoreV1().RESTClient()
+	deploymentsClient := kubeClient.AppsV1().Deployments(apiv1.NamespaceDefault)
+	resyncPeriod := 30 * time.Second
+	listWatch := k8sCache.NewListWatchFromClient(crdClient, "messagequeuetriggers", metav1.NamespaceAll, fields.Everything())
+	_, controller := k8sCache.NewInformer(listWatch, &fv1.MessageQueueTrigger{}, resyncPeriod, k8sCache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			go func() {
+				mqt := obj.(*fv1.MessageQueueTrigger)
+				logger.Debug("Create deployment for Scaler Object", zap.Any("mqt", mqt.ObjectMeta), zap.Any("maqt.Spec", mqt.Spec))
+				scaledObject := getScaledObject(mqt)
+				kedaClient, err := getKedaClient(mqt.ObjectMeta.Namespace)
+				if err != nil {
+					logger.Error("Failed to create KEDA client", zap.Error(err))
+				}
+				_, err = kedaClient.Create(scaledObject, metav1.CreateOptions{})
+				if err != nil {
+					logger.Error("Failed to create ScaledObject", zap.Error(err))
+				}
```

Comment:
If the scaled object can not be created - we should not create the deployment as well. Same would be case other way around too

Reply:
Done